app/src/main/java/de/westnordost/streetcomplete/data/osm/osmquests/OsmQuestController.kt

```diff
@@ -151,7 +153,9 @@ class OsmQuestController internal constructor(
             scope.async {
                 val questsForType = ArrayList<OsmQuest>()
                 val questTypeName = questType::class.simpleName!!
-                if (!countryBoundaries.intersects(bbox, questType.enabledInCountries)) {
+                val countryCodes = countryBoundaries.getIntersectingIds(bbox)
+
+                if (!countryCodes.any { questType.isEnabled(countryInfos.get(listOf(it))) }) {
```

Comment:
The corresponding test (["updates quests on map data listener replace for bbox"](https://github.com/streetcomplete/StreetComplete/blob/master/app/src/test/java/de/westnordost/streetcomplete/data/osm/osmquests/OsmQuestControllerTest.kt#L299)) is now failing. I'm not sure whether the test needs to be adapted or I accidentally changed the logic here.